lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -728,20 +729,22 @@ static bool areBundleElementsEquivalent(BundleType::BundleElement destElement,
                                         BundleType::BundleElement srcElement,
                                         bool destOuterTypeIsConst,
                                         bool srcOuterTypeIsConst,
-                                        bool requiresSameWidth) {
+                                        bool requiresSameWidth,
+                                        bool requireDestSameOrUninferred) {
   if (destElement.name != srcElement.name)
     return false;
   if (destElement.isFlip != srcElement.isFlip)
     return false;
 
   if (destElement.isFlip) {
+    assert(!requireDestSameOrUninferred);
```

Comment:
I see why this is here, but this incompatibility should at least be documented on the function.

Reply:
Heh, the method documentation is really out-of-date, mentions flip canonicalization (and the method taking only passive types) :sweat_smile: .  Sounds good to refresh for both reasons, thanks!